daft/ai/transformers/provider.py

```diff
@@ -63,6 +86,17 @@ def get_text_embedder(
         return TransformersTextEmbedderDescriptor(model or self.DEFAULT_TEXT_EMBEDDER, options)
 
     def get_image_classifier(self, model: str | None = None, **options: Any) -> ImageClassifierDescriptor:
+        from daft.dependencies import pil_image, torchvision
+
+        if not torchvision.module_available():
+            raise ImportError(
+                "torchvision is required to classify_image with TransformersProvider. Please install it using `pip install 'daft[transformers]'`"
+            )
+        if not pil_image.module_available():
+            raise ImportError(
+                "Pillow is required to classify_image with the TransformersProvider. Please install it using `pip install 'daft[transformers]'`"
+            )
```

Comment:
All this should use the ProviderImportError which creates this message for you, you could add a 'function' parameter if you want to.